fuzz/fuzz_targets/hexf_parse.rs

```diff
@@ -0,0 +1,78 @@
+#![no_main]
+
+use std::{ffi::CString, ptr};
+
+use errno::{errno, set_errno, Errno};
+use libc::strtod;
+use libfuzzer_sys::{arbitrary::Arbitrary, fuzz_target};
+
+use hexf_parse::{parse_hexf32, parse_hexf64};
+
+#[derive(Debug, Arbitrary)]
+enum TargetType {
+    F32,
+    F64,
+}
+
+// IMPORTANT: For the verification with strtof/d to work properly, your locale (LC_ALL) should be set C.UTF-8.
+//            Otherwise strtof/d will use your locale's decimal separator.
+
+macro_rules! verify_with_strto_fn {
+    ($target_name:ident, $strto_fn:ident, $string_to_parse:ident, $result:ident) => {
+        if $string_to_parse.contains("p") || $string_to_parse.contains("P") {
+            if let Ok(c_string) = CString::new($string_to_parse.replace('_', "")) {
+                unsafe {
+                    // strtod will write a *mut c_char to the character past the last interpreted character in here
+                    let mut end_ptr = ptr::null_mut();
+
+                    let strto_fn_result = $strto_fn(c_string.as_ptr(), &mut end_ptr);
```

Comment:
This should set errno to zero before calling `$strto_fn` in order to reliably detect erors, as described in eg. [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/functions/strtod.html): "an application wishing to check for error situations should set errno to 0, then call strtod(), strtof(), or strtold(), then check errno."